core/src/main/java/com/amplifyframework/core/model/query/predicate/QueryPredicateGroup.java

```diff
@@ -95,7 +95,9 @@ public QueryPredicateGroup or(QueryPredicate predicate) {
      * Return a group negating the given group of operations.
      * @param predicate the group to negate
      * @return a group negating the given group of operations
+     * @deprecated use {@link QueryPredicate#not(QueryPredicate)} instead.
      */
+    @Deprecated
```

Comment:
I don't think it makes sense to have multiple copy-pasted code scattered throughout the codebase. Deprecated code is just an admittance that we didn't think this through at the time of implementing this, and I don't see that as necessarily a bad thing

Reply:
I wasn't suggesting we have multiple implementations.  I was suggesting we just keep the one here in `QueryPredicateGroup`, and not add a new one in `QueryPredicate`.  What do you think?

Reply:
ah sorry i must have misremembered it. You are absolutely correct! I'll do it as you suggested.

Reply:
`QueryPredicateOperation` also has the same implementation, which suggests that it would be better to simply move the whole thing to the interface and deprecate the individual implementations in children?

Reply:
the same static method currently exists in both `QueryPredicateGroup` and `QueryPredicateOperation`, so I think it makes sense to move it up to the interface and deprecate the currently existing ones.

Reply:
> the same static method currently exists in both `QueryPredicateGroup` and `QueryPredicateOperation`, so I think it makes sense to move it up to the interface and deprecate the currently existing ones.

Why?  Seems like a breaking change with no customer benefit and a trivial benefit to us (our code is a little more DRY)